central/tlsconfig/tlsconfig.go

```diff
@@ -56,39 +57,54 @@ func GetAdditionalCAs() ([][]byte, error) {
 	return certDERs, nil
 }
 
-// GetDefaultCertChain reads and parses default cert chain and returns it in DER encoded format
-func GetDefaultCertChain() ([][]byte, error) {
-	certFile := filepath.Join(DefaultCertPath, TLSCertFileName)
-	content, err := os.ReadFile(certFile)
+// MaybeGetDefaultCertChain reads and parses default cert chain and returns it in DER encoded format
+func MaybeGetDefaultCertChain() ([][]byte, error) {
+	cert, err := MaybeGetDefaultTLSCertificateFromDirectory(DefaultCertPath)
 	if err != nil {
-		// Ignore error if default certs do not exist on filesystem
-		if os.IsNotExist(err) {
-			return nil, nil
-		}
-		return nil, errors.Wrap(err, "reading default cert file")
+		return nil, err
 	}
-
-	certDERsFromFile, err := x509utils.ConvertPEMToDERs(content)
-	if err != nil {
-		return nil, errors.Wrap(err, "converting additional CA cert to DER")
+	if cert == nil {
+		return nil, nil
 	}
+	return cert.Certificate, nil
+}
 
-	return certDERsFromFile, nil
+// MaybeGetDefaultTLSCertificateFromDefaultDirectory load the default tls certificate from the default directory.
+func MaybeGetDefaultTLSCertificateFromDefaultDirectory() (*tls.Certificate, error) {
+	return MaybeGetDefaultTLSCertificateFromDirectory(DefaultCertPath)
 }
 
-// loadDefaultCertificate load the default tls certificate
-func loadDefaultCertificate(dir string) (*tls.Certificate, error) {
+// MaybeGetDefaultTLSCertificateFromDirectory load the default tls certificate from the given directory.
+func MaybeGetDefaultTLSCertificateFromDirectory(dir string) (*tls.Certificate, error) {
 	certFile := filepath.Join(dir, TLSCertFileName)
 	keyFile := filepath.Join(dir, TLSKeyFileName)
 
-	if filesExist, err := fileutils.AllExist(certFile, keyFile); err != nil || !filesExist {
-		return nil, err
+	if exists, err := fileutils.Exists(certFile); err != nil || !exists {
+		if err != nil {
+			log.Warnw("Error checking if default TLS certificate file exists", zap.Error(err))
+			return nil, err
+		}
+		log.Infof("Default TLS certificate file %q does not exist. Skipping", certFile)
+		return nil, nil
+	}
+
+	if exists, err := fileutils.Exists(keyFile); err != nil || !exists {
+		if err != nil {
+			log.Warnw("Error checking if default TLS key file exists", zap.Error(err))
+			return nil, err
+		}
+		log.Infof("Default TLS key file %q does not exist. Skipping", keyFile)
+		return nil, nil
 	}
 
 	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
 	if err != nil {
-		return nil, err
+		if strings.Contains(err.Error(), "private key does not match public key") {
+			return nil, errors.Wrap(err, "loading default certificate. If the certificate file contains a certificate chain, ensure that the certificate chain is in the correct order (the first certificate should be the server certificate, any following certificates should form the certificate chain).")
```

Comment:
IIRC error messages should not end with a full stop...
```suggestion
			return nil, errors.Wrap(err, "loading default certificate failed; if the certificate file contains a certificate chain, ensure that the certificate chain is in the correct order (the first certificate should be the server certificate, any following certificates should form the certificate chain)")
```

Reply:
Also, would it be better to say `leaf certificate` instead of `server certificate`? I think there is nothing preventing use of this function to load client certificates, where mentioning "server" would be rather confusing...